napi_js_props.md

```diff
@@ -0,0 +1,437 @@
+[Section 6.1.7.1](https://tc39.github.io/ecma262/#table-2) of the
+[ECMAScript Language Specification](https://tc39.github.io/ecma262/). They can
+be one or more of the following bitflags:
+
+- `napi_default` - Used to indicate that no explicit attributes are set on the 
+given property. By default, a property is Writable, Enumerable, and
+ Configurable. Note that this is a deviation from the ECMAScript specification,
+ where generally the values for a property descriptor attribute default to 
+ false if they're not provided.
+- `napi_read_only` - Used to indicate that a given property is not Writable
+- `napi_dont_enum` - Used to indicate that a given property is not Enumerable
+- `napi_dont_delete` - Used to indicate that a given property is not 
+Configurable, as defined in 
+[Section 6.1.7.1](https://tc39.github.io/ecma262/#table-2) of the 
+[ECMAScript Language Specification](https://tc39.github.io/ecma262/).
+- `napi_static_property` - Used to indicate that the property will be defined as
+a static property on a class as opposed to an instance property, which is the
+default. This is used only by `napi_define_class`. It is ignored by 
+`napi_define_properties`
+
+### *napi_property_descriptor*
+
+#### Definition
+```C
+typedef struct {
+  const char* utf8name;
+
+  napi_callback method;
+  napi_callback getter;
+  napi_callback setter;
+  napi_value value;
+
+  napi_property_attributes attributes;
+  void* data;
+} napi_property_descriptor;
+```
+
+#### Members
+- `utf8name`: String describing the key for the property, encoded as UTF8
+- `value`: The value that's retrieved by a get access of the property if the
+ property is a data. If this is passed in, set `getter`, `setter`, `method` 
+ and `data` to `NULL` (since these members won't be used)
+- `getter`: A function to call when a get access of the property is performed.
+If this is passed in, set `value` and `method` to `NULL` (since these members
+won't be used). The given function is called implicitly by the runtime when the
+property is accessed from JavaScript code (or if a get on the property is 
+performed using a N-API call)
+- `setter`: A function to call when a set access of the property is performed.
+If this is passed in, set `value` and `method` to `NULL` (since these members
+won't be used). The given function is called implicitly by the runtime when the
+property is set from JavaScript code (or if a set on the property is 
+performed using a N-API call).
+- `method`: Set this if you want the property descriptor object's `value`
+property to be a JavaScript function represented by `method`. If this is
+passed in, set `value`, `getter` and `setter` to `NULL` (since these members
+won't be used)
+- `data`: The callback data passed into `method`, `getter` and `setter` if 
+this function is invoked
+- `attributes`: The attributes associated with the particular property. 
+See `napi_property_attributes`
```

Comment:
`napi_property_attributes` is also documented in this file, so this would be an internal anchor.